build-logic/publishing/src/main/kotlin/gradlebuild.publish-public-libraries.gradle.kts

```diff
@@ -159,6 +159,6 @@ tasks.register("promotionBuild") {
     group = "publishing"
     dependsOn(
         ":distributions-full:verifyIsProductionBuildEnvironment", ":distributions-full:buildDists", ":distributions-full:copyDistributionsToRootBuild",
-        ":distributions-integ-tests:forkingIntegTest", ":docs:releaseNotes", "publish", ":docs:incubationReport", ":docs:checkDeadInternalLinks"
+        ":distributions-integ-tests:forkingIntegTest", ":docs:releaseNotes", "publish", ":docs:incubationReport", ":docs:checkDeadInternalLinks", ":docs:checkSamples"
```

Comment:
We should not have this flakiness in the promotion build. That would mean that the actual promotion would fail more often, which would be very bad. Can we move the check to the ready for release stage instead?

I also don't think that `checkSamples` is the right task to move: We only want to move the link test, not all of `checkSamples`, right? I currently don't know what other checks there are, though maybe those checks are not depending on the Java version, so we could move them to the distribution build.

Reply:
I did check the source code of guides plugin and it seems not possible to only remove checklinks unless use something like:

```
tasks.withType(CheckLinks) {
  enabled = findProperty("checkLinks") == "true"
}
```

> we could move them to the distribution build.

Moving to distribution build doesn't solve the problem that frequently fails our feedback pipeline.

`checkSamples` are a bit more than just check links, but I think it's better than using `-PcheckLinks=true`. Maybe it belongs to ReadyForRelease.

Reply:
I meant moving everything apart from the link checking to the distribution build. The link checking should go in a job in ready for release. Since I don't think we have an existing job in ready for release already, it probably needs to be a new one.